### ISSUE TITLE
-- ERROR: Error: The parameter 'WorkingSetPrimType' is not recognized. -RizomUV

### ISSUE THREAD

**Issue by NicholasWoudman** [Smithsonian/dpo-cook#75]
I wanted to use the unwrapping feature of RizomUV but every time I run the recipe it give me this error ''-- ERROR: Error: The parameter 'WorkingSetPrimType' is not recognized." inside Rizom.

In the tool section of Rizom inside Cook (RizomUVTool.ts) there is a constant called Content with the line `ZomSelect({PrimType="Edge", Select=true, ResetBefore=true, WorkingSetPrimType="Island"... (Rest of code)'` 

I tried removing the WorkingSetPrimType section to see what it changed but the exact same error still continued.

So I went I digging in Rizom's documentation and I found out that WorkingSet and PrimType are both separate parameters. PrimType='Islands' is correct  (Or the values ''Vertex''', ''Edge'', "Triangle'', ''Polygon'') but WorkingSet expects "Visible", "Selected", "Flat", "NotFlat". But after changing the line in RizomUVTool.ts to have it include the fixed parameter it doesn't update the error. Inside Rizom it keeps saying that the given script includes the incorrect line even though I checked everywhere for any appearance of that line. I'm kinda stuck because I don't understand why Cook keeps passing the wrong line of code to Rizom.

Any help?

**Comment by gjcope** [comment]
Hi @NicholasWoudman - A couple of issues here I think. The [current version of RizomUV we are supporting](https://smithsonian.github.io/dpo-cook/tools/rizom-uv/) is really old. I believe since then they have slightly changed the api (as you noted) so we have some work to do to get current. As far as making those updates yourself, after changing RizomUVTool.ts are you then rebuilding the app and relaunching the server?

**Comment by NicholasWoudman** [comment]
Thanks for the quick reply @gjcope, what a legend

Ofcourse it started working after I posted the issue, very typical haha. I first ran it with a 'npm run server' but that didn't work, after I launched it in development mode it started using the changes I made. Now run server also works like intended. I am not sure what I did differently or why it now works but it seems not to error anymore after fixing the parameter and started unfolding (very slowly I might add but I think that's more due to my college laptop and probably not setting a lower variable in the recipe, when click the unfold button inside Rizom it works much faster but it creates a single flat so ill keep experimenting and figuring stuff out).

**Comment by gjcope** [comment]
Great news, glad it's working now. Please do update on anything you learn with the speed difference. On our documentation page it says we are using an old version because there was a bug in newer ones when running from script. But this was before I joined the project so I'm not actually sure what that bug was! Maybe it has something to do with this slow down.

**Comment by NicholasWoudman** [comment]
@gjcope 
welp, I was too quick to celebrate. The slow down is indeed the bug you mentioned, I had it run for almost an hour an it was stuck on 6% the entire time, I tried messing around in the files myself but alas nothing worked. When I installed the 2018 version then everything worked flawlessly so yea this entire thread is kinda useless haha oh well. Atleast we know that its still broken lol
